### PR TITLE
Backend question

### LINKED ISSUE #3: Question Service
## Description
- Each question must have the following details:
  - Id
  - Title
  - Description
  - Category
  - Complexity
- Can refer to https://github.com/CS3219-AY2425S1/PeerPrep-UserService/tree/main

### PR DESCRIPTION
**Pull request by jeraldkiew**
1. Adds question service with APIs for CRUD of questions.
2. Add question mongodb database

Md file API.md is included for documentation of the APIs

closes issue #3

### CODE REVIEW

File: question-service/controller/create.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        id: id,
+        title: title,
+        description: description,
+        topics: topics,
+        difficulty: difficulty,
+      });
+      await question.save();
+      return res.status(200).json(question);;
+    } catch (err) {
+      console.log(err);
```
Comment (naozee) [inline question-service/controller/create.js]: Remove console log. Add to message region & don't put unknown.

Comment (naozee) [reply question-service/controller/create.js]: Try to do for all errors

File: question-service/controller/read.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    const questions = await Question.find({ 
+      topics: topic
+    });
+    return res.status(200).json(questions);;
+  } catch (err) {
+    console.log(err);
+    return res.status(500).json({ message: `Unknown error when finding question with topic ${topic}!` });
+  }
+};
+
```
Comment (naozee) [inline question-service/controller/read.js:83]: Add multiple filters (e.g. by difficulty and topic)

**Review by naozee — COMMENTED** [review]
Add docker file pls :)

**Review by naozee — APPROVED** [review]
lgtm

**Review by jasonqiu212 — APPROVED** [review]
lgtm

**Pull request merged**